packages/react-pages/src/node/dynamic-modules/pages.ts

```diff
@@ -211,14 +201,109 @@ function createFindPagesContext(): [FindPagesResult, FindPagesHelpers] {
     }
   }
 
+  const pageIdByFilePath: { [filePath: string]: string } = {}
+  const watchFiles = (baseDir: string, glob: string) => {
+    // Strip trailing slash and make absolute
+    baseDir = path.resolve(baseDir)
+
+    let watcher = watchers[baseDir]
+    if (!watcher) {
+      const reloadPageData = async (filePath: string) => {
+        const relative = filePath.slice(baseDir.length + 1)
+        const pageData = await loadPageData(
+          { relative, absolute: filePath },
+          helpers
+        )
+        pageIdByFilePath[filePath] = pageData.pageId
+        addPageData(pageData)
+      }
+
+      watchers[baseDir] = watcher = chokidar.watch([], {
+        cwd: baseDir,
+        ignored: ['**/node_modules/**/*'],
+        ignoreInitial: true,
+      })
+
+      watcher
+        .on('add', async (filePath) => {
+          filePath = path.join(baseDir, filePath)
+          addResult(reloadPageData(filePath))
```

Comment:
What if the file added is not a page file? For example, baseDir/utils.ts is added.

Reply:
How about add a `isPageFile(filePath): boolean` to the `PageStrategy`? So that user can tell us that `baseDir/utils.ts` is not a page file?

Reply:
Here is how vite match "added file" with "glob" ([glob import feature](https://vitejs.dev/guide/features.html#json)): 
https://github.com/vitejs/vite/blob/c323732dca0a439bd0477da660a6a516289b3ce9/packages/vite/src/node/server/hmr.ts#L172

Reply:
In my opinion, `globFind` should be dedicated to finding pages. If the `PageStrategy` wants to search the filesystem for non-page files, it can use `globby` directly.

Also, we could add a `watchPage(filePath)` helper for watching pages not found through the `globFind` helper.

The `isPageFile` idea seems unappealing, because then the `PageStrategy` has to repeat its logic for identifying pages (once in `isPageFile`, again in `findPages`).

Reply:
I am not sure I understand you correctly. In the above example about `baseDir/utils.ts`. When `baseDir/utils.ts` is created, we **do not** want to process it like a page file. But in your current code, it is treated like a page file. Isn't it a bug? That is what I am talking about.

> The isPageFile idea seems unappealing, because then the PageStrategy has to repeat its logic for identifying pages (once in isPageFile, again in findPages).

Yes the logic is indeed repeated. I don't have a better idea for now :)

Reply:
> But in your current code, it is treated like a page file. 

Using the default strategy? That doesn't sound right, as the default strategy uses this glob: `**/*$.{md,mdx,js,jsx,ts,tsx}` which would never match that file.

If you're using a custom strategy, use `globby` directly, instead of `globFind`.

> I don't have a better idea for now

The better idea, in my opinion, is to reserve `globFind` for page-related globs only.

Reply:
> If you're using a custom strategy, use globby directly, instead of globFind. If you do that, you can filter the glob matches as you see fit, then pass the page-related matches to the watchPage helper (not yet implemented).

If users are using a custom strategy and need to do custom filtering, they need to use chokidar, right? Because globby doesn't provide file watching ability.

Reply:
We can avoid direct use of `chokidar` by providing these helpers:
- `watchFiles(baseDir: string, globs: string | string[], cb: Function)`
  - where `cb` is `(file: string, type: 'add' | 'change' | 'unlink') => void`
  
- `setPageData(pageId: string, pageData: object | null)`
  - where `pageId` should be the filename (or another string if the page was generated some other way)
  - pass null to delete page

Then we can go back to `PageStrategy` being a single function...
```
(pagesDir: string, helpers: FindPagesHelpers) => void
```
...instead of `{ findPages, loadPageData }`.

Reply:
> where cb is `(file: string, type: 'add' | 'change' | 'unlink') => void`

I think the 'change' type is not need. File update has already been tracked by vite hmr.

Reply:
> Users don't need to handle it because we can delete page for users.

That depends on whether we want to support generated pages (ie: pages not associated one-to-one with a specific file). For example, I may want to generate a page that links to a subset of file-based pages.

Reply:
> That depends on whether we want to support generated pages (ie: pages not associated one-to-one with a specific file). For example, I may want to generate a page that links to a subset of file-based pages.

Vite-pages already support composed page. User can associate one page with multi pageFiles, identified by `key`: 
```js
helpers.addPageData({
  pageId,
  key: pageDataKey,
  dataPath: pageFilePath,
  staticData: await helpers.extractStaticData(absolute),
})
```
Checkout [this fixture](https://github.com/vitejs/vite-plugin-react-pages/blob/db37207d047fed8d9ef02f07750eed49eba956ad/fixtures/custom-find-pages/vite.demos.ts#L31). Two demos of `Button` is rendered in same page.

I think **users don't need to handle file deletion even if they want to have composed page**. Because when users call this:
```js
helpers.addPageData({
  pageId,
  key: pageDataKey,
  dataPath: pageFilePath,
  staticData: await helpers.extractStaticData(absolute),
})
```
Vite-page can handle the deletion of `pageFilePath`, deleting the pageData named `pageDataKey` from the page. Users don't need to take care of this.

Reply:
And what if `dataPath` is undefined?